### ISSUE TITLE
does not report error on combobox associated popup with incorrect role

### ISSUE THREAD

**Issue by Zidious** [dequelabs/axe-core#3217]
Ref: https://github.com/dequelabs/axe-core-npm/issues/314

Example given:
```js
 <div aria-controls="associated-element"
       role="combobox"
       aria-label="foo"
       aria-expanded="true"
       aria-haspopup="listbox"
       >
  </div>
  <div
       role="alert"
       id="associated-element"
       aria-label="bar">
  </div>
  ```
